File: liblua/len.lua
#!/usr/bin/lua

function main()
    if #arg == 0 then
        print("Missing Arguments")
        return 1
    end

    local chars, words = 0, 0
    for i = 1, #arg do
        io.write("["..i.."] " .. arg[i].." -> " .. #arg[i] .. "\n")
        chars = chars + #arg[i]
        words = words + 1
    end
    io.write("\nWords -> " .. words)
    io.write("\nCharacters -> " .. chars .. "\n")
end

main()

File: libgo/len.go
package main

import (
	"fmt"
	"os"
)

func main() {
	if len(os.Args) < 2 {
		fmt.Fprintf(os.Stderr, "Missing Parameters!\n")
		os.Exit(1)
	}
	words, chars := 0, 0
	for i := 1; i < len(os.Args); i++ {
		words++
		fmt.Printf("[%d] %s -> %d\n", i, os.Args[i], len(os.Args[i]))
		chars += len(os.Args[i])
	}
	fmt.Printf("\nWords -> %d\n", words)
	fmt.Printf("Characters -> %d\n", chars)
}

